### PR TITLE
[ISSUE #891] need lock service when create instance

### LINKED ISSUE #891: 存储层新增实例时，批处实现没有对服务加锁
**Describe the bug**

存储层的addInstance方法中通过rlockServiceWithID对服务加锁了
但在batchAddInstances批量方法中却没有对服务加锁

### PR DESCRIPTION
**Pull request by chuntaojun**
**Please provide issue(s) of this PR:**
Fixes #891 

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] ApiServer
- [ ] Auth
- [ ] Configuration
- [x] Naming
- [ ] HealthCheck
- [ ] Metrics
- [ ] Docs
- [x] Installation
- [ ] Performance and Scalability
- [x] Test and Release

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any user-facing changes. This may include API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Review by daheige — APPROVED** [review]

**Review by houseme — APPROVED** [review]

**Review by shichaoyuan — APPROVED** [review]

File: service/batch/instance.go [hunk 1]
```diff
@@ -287,26 +287,41 @@ func (ctrl *InstanceCtrl) registerHandler(futures []*InstanceFuture) error {
...
+	unlockService := make(map[string]func())
+	for i := range remains {
+		svcId := remains[i].serviceId
+		if _, ok := unlockService[svcId]; ok {
+			continue
+		}
+		_, releaseFunc, err := ctrl.lockServiceByID(context.Background(), svcId)
+		if err != nil {
+			sendReply(futures, apimodel.Code_StoreLayerException, err)
+			continue
```
Comment (andrewshan) [inline service/batch/instance.go:314]: 这里已经sendReply了？为啥还要continue？

File: service/batch/instance.go [hunk 2]
```diff
@@ -287,26 +287,41 @@ func (ctrl *InstanceCtrl) registerHandler(futures []*InstanceFuture) error {
...
 	// 构造model数据
 	for _, entry := range remains {
 		ins := instancecommon.CreateInstanceModel(entry.serviceId, entry.request)
-		if _, ok := firstRegisInstances[ins.ID()]; ok {
-			ins.FirstRegis = true
-		}
 		entry.SetInstance(ins)
 	}
 
+	unlockService := make(map[string]func())
```
Comment (andrewshan) [inline service/batch/instance.go:305]: 在这里统一进行transaction的创建

File: service/batch/instance.go [hunk 3]
```diff
@@ -443,80 +458,66 @@ func (ctrl *InstanceCtrl) batchRestoreInstanceIsolate(futures map[string]*Instan
...
+		return err
 	}
 
-	firstRegisInstances := make(map[string]struct{})
 	if len(id2Isolate) == 0 {
-		for id := range futures {
-			firstRegisInstances[id] = struct{}{}
-		}
-		return firstRegisInstances, err
+		return err
```
Comment (andrewshan) [inline service/batch/instance.go:465]: 这里改成return nil吧

File: service/batch/instance.go [hunk 4]
```diff
@@ -443,80 +458,66 @@ func (ctrl *InstanceCtrl) batchRestoreInstanceIsolate(futures map[string]*Instan
...
-		}
-
 		for id, isolate := range id2Isolate {
 			if future, ok := futures[id]; ok && future.request.Isolate == nil {
 				future.request.Isolate = &wrappers.BoolValue{Value: isolate}
 			}
 		}
 	}
-	return firstRegisInstances, err
+	return err
```
Comment (andrewshan) [inline service/batch/instance.go:475]: 改成return nil

File: service/batch/instance.go (hunk 1 above)
Comment (chuntaojun) [reply service/batch/instance.go:322]: > 这里已经sendReply了？为啥还要continue？

调整了，之前的逻辑有问题，这里是需要单个 reply

**Review by andrewshan — APPROVED** [review]
LGTM

**Pull request merged**